### PR TITLE
Add test that runs the S3 scrubber

### LINKED ISSUE #6080: Test scrubber with initdb files
A test should check:
- That a regular active tenant is not reported as an error due to having an initdb file
- That a post-deletion tenant is not reported as an error due to its remaining initdb file

Look for where S3Scrubber is already used for examples.

Testing gap was noticed while quickly fixing https://github.com/neondatabase/neon/pull/6079 to use the scrubber.

### PR DESCRIPTION
**Pull request by arpad-m**
In #6079 it was found that there is no test that executes the scrubber. We now add such a test, which does the following things:

* create a tenant, write some data
* run the scrubber
* remove the tenant
* run the scrubber again

Each time, the scrubber runs the scan-metadata command. Before #6079 we would have errored, now we don't.

Fixes #6080

### CODE REVIEW

**Comment by arpad-m** [comment]
tests fail due to missing control plane API. Obviously we don't have that one around. Should I add a mode that allows one to ignore the control plane, disabling the "does tenant exist check", and assuming it always exists?

File: test_runner/regress/test_tenant_delete.py [hunk 1]
```diff
@@ -669,3 +671,37 @@ def deletion_arrived():
...
+
+    scrubber.scan_metadata()
+
+    env.start()
+    ps_http = env.pageserver.http_client()
+    iterations = poll_for_remote_storage_iterations(remote_storage_kind)
+    tenant_delete_wait_completed(ps_http, tenant_id, iterations)
+    env.stop()
+
+    scrubber.scan_metadata()
```
Comment (jcsp) [inline test_runner/regress/test_tenant_delete.py:707]: Can you add some assertions after scan_metadata calls that check the output?  It returns a dict with the json output from the  tool.

`scan_metadata` already fails on issues that the scrubber thinks are fatal (the `with_errors` timeline list in the output), but for a test like this we can be stricter and also assert that with_warnings is empty.

**Review by jcsp — APPROVED** [review]
LGTM, just a suggestion to make the test a little stricter.

File: test_runner/regress/test_tenant_delete.py (hunk 1 above)
Comment (arpad-m) [reply test_runner/regress/test_tenant_delete.py:708]: Added such an assert.

**Pull request merged**